v1/message_test.go

```diff
@@ -20,35 +21,90 @@ var (
 )
 
 func TestDecodeMessage(t *testing.T) {
-	val, err := DecodeMessage(testMessageBytes)
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	val, err := DecodeMessage(&key.PublicKey, testMessageBytes)
 	require.NoError(t, err)
 	require.EqualValues(t, StatusMessage{
-		Message: testMessageStruct,
+		Message:   testMessageStruct,
+		SigPubKey: &key.PublicKey,
+		ID:        MessageID(&key.PublicKey, testMessageBytes),
```

Comment:
Should we test that it fails if we pass a different pubkey (just to ensure that it's indeed taken into account)?

Reply:
Good idea, done, I have added a test for `MessageID`